sudokuGUI.py

```diff
@@ -1,532 +1,533 @@
+            if event.type == pygame.MOUSEBUTTONDOWN:
+                for button in self.menu_buttons:
+                    if button.highlighted:
+                        button.click()    
+
+
+    # Function that updates the game (clicking on buttons etc)
+    # Input: nothing
+    # Output: Nothing
+    def menu_update(self):
+        self.mouse_pos = pygame.mouse.get_pos()
+        for button in self.menu_buttons:
+            button.update(self.mouse_pos)
+
+
+    # Function that draws the menu with the levels option
+    # Input: Nothing
+    # Output: Nothing
+    def menu_draw(self):
+        self.window.fill(BLACK)
+
+        for button in self.menu_buttons:
+            button.draw(self.window)
+  
+        menu_font = pygame.font.SysFont('comicsans', 100)
+        font = menu_font.render(self.screen_text, False, PINK)
+        self.window.blit(font, [65, 50])
+
+        pygame.display.update()
+        pygame.display.set_caption(self.caption)
+
+
+    # Function that randomly gets a board from the DataBase class and initiate variables by it
+    # Input: The level the user chose
+    # Output: Nothing
+    def get_board(self, level : int):
+        self.grid = self.db.get_rand_board(level)
+        self.finished_board = deepcopy(self.grid)
+        solving(self.finished_board)
+        self.load_board()
+        self.state = 'playing'
+        self.start = time.time()
+
+# Playing state functions # 
+
+    # Function that responsible on the playing differen events
+    # Input: Nothing
+    # Output: Nothing
+    def playing_events(self):
+        for event in pygame.event.get():
+            if event.type == pygame.QUIT:
+                self.running = False
+            
+            # When the user clicks
+            if event.type == pygame.MOUSEBUTTONDOWN:
+                if self.mouse_pos:
+                    self.error_msg = False
+                    selected = self.mouse_on_grid()
+
+                if selected:
+                    self.selected = selected
+                else:
+                    self.selected = None
+                    for button in self.playing_buttons:
+                        if button.highlighted:
+                            button.click()
+                            break
+            
+            # When the user type
+            if event.type == pygame.KEYDOWN:
+                if self.selected != None and self.selected not in self.lock_cells: # Checking if the position is a blank rect
+                    if self.is_int(event.unicode): # Checking if the user entered a number 
+                        self.temp_values[(self.selected[0], self.selected[1])] =  int(event.unicode) # temp_values[(x, y)] = num
+                    
+                    elif event.key == pygame.K_BACKSPACE:
+                        del self.temp_values[(self.selected[0], self.selected[1])]
+
+                if event.key == pygame.K_KP_ENTER:
+                    pos = (self.selected[0], self.selected[1])
```

Comment:
I would move this up because there are more places where you use `(self.selected[0], self.selected[1])` to avoid code duplication